### PR TITLE
Feature/oh my class/oh my class#22 entity structure as representation for database

### PR DESCRIPTION
**Pull request by maugra**
ready for review

### CODE REVIEW

File: src/main/java/com/ohmyclass/api/components/classmember/controller/IClassMember.java [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package com.ohmyclass.api.components.classmember.controller;
+
+public interface IClassMember {}
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/api/components/classmember/controller/IClassMember.java]: Rename to IClassMemberController

File: src/main/java/com/ohmyclass/api/components/classmember/controller/impl/ClassMember.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package com.ohmyclass.api.components.classmember.controller.impl;
+
+public class ClassMember {
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/api/components/classmember/controller/impl/ClassMember.java]: Rename to ClassMemberController

Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/classmember/controller/impl/ClassMember.java]: Should probably be called **GroupMemberController** for better readability and comprehensability.

File: src/main/java/com/ohmyclass/api/components/classmember/controller/IClassMember.java (hunk 1 above)
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/classmember/controller/IClassMember.java]: Should probably be called **IGroupMemberController** for better readability and comprehensability.

File: src/main/java/com/ohmyclass/api/components/classmember/entity/ClassMember.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+@Table(name = "classmember")
+public class ClassMember {
+
+    @Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	@Column(name = "id", nullable = false)
+	private Long id;
+
+	@ManyToMany(fetch = FetchType.EAGER)
+	private Set<User> user;
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/api/components/classmember/entity/ClassMember.java]: Why is this a set? This is _the_ mc:mc table.
(Should look like fkGroup)

File: src/main/java/com/ohmyclass/api/components/classmember/entity/ClassMember.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import com.ohmyclass.api.components.user.entity.User;
+import com.ohmyclass.api.components.role.entity.Role;
+
+@Getter
+@Setter
+@Entity
+@Table(name = "classmember")
+public class ClassMember {
+
+    @Id
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/classmember/entity/ClassMember.java:19]: One tab too much.

File: src/main/java/com/ohmyclass/api/components/classmember/entity/ClassMember.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+	@ManyToMany(fetch = FetchType.EAGER)
+	private Set<User> user;
+
+	@ManyToOne
+	@JoinColumn(name = "fkGroup")
+	@JsonManagedReference
+	private Group fkGroup;
+
+//	@ManyToMany(fetch = FetchType.EAGER)
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/api/components/classmember/entity/ClassMember.java]: Will be added eventually

File: src/main/java/com/ohmyclass/api/components/comment/entity/Comment.java [hunk 6]
```diff
@@ -1,3 +1,53 @@
...
+	private long id;
+
+	@Column(name = "content")
+	private String content;
+
+	@Column(name = "date")
+	private Date date;
+
+	@ManyToOne
+	private Comment fkParentComment;
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/comment/entity/Comment.java]: Attribute name shouldn't contain **fk**, only the column name in the database.

File: src/main/java/com/ohmyclass/api/components/classmember/entity/ClassMember.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+import java.util.List;
+
+import com.ohmyclass.api.components.user.entity.User;
+import com.ohmyclass.api.components.role.entity.Role;
+
+@Getter
+@Setter
+@Entity
+@Table(name = "classmember")
+public class ClassMember {
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/api/components/classmember/entity/ClassMember.java:17]: Rename to GroupMember. The term "class" does not exist in this context

File: src/main/java/com/ohmyclass/api/components/classmember/dto/out/ClassMemberOutDTO.java [hunk 8]
```diff
@@ -0,0 +1,3 @@
+package com.ohmyclass.api.components.classmember.dto.out;
+
+public class ClassMemberOutDTO {}
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/api/components/classmember/dto/out/ClassMemberOutDTO.java]: Rename to GroupMemberOutDTO

File: src/main/java/com/ohmyclass/api/components/classmember/dto/in/ClassMemberInDTO.java [hunk 9]
```diff
@@ -0,0 +1,3 @@
+package com.ohmyclass.api.components.classmember.dto.in;
+
+public class ClassMemberInDTO {}
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/api/components/classmember/dto/in/ClassMemberInDTO.java]: Rename to GroupMemberInDTO

File: src/main/java/com/ohmyclass/api/components/classmember/controller/impl/ClassMember.java (hunk 2 above)
Comment (Z-100) [reply src/main/java/com/ohmyclass/api/components/classmember/controller/impl/ClassMember.java]: *GroupMemberController

File: src/main/java/com/ohmyclass/api/components/classmember/controller/IClassMember.java (hunk 1 above)
Comment (Z-100) [reply src/main/java/com/ohmyclass/api/components/classmember/controller/IClassMember.java]: *IGroupMemberController

File: src/main/java/com/ohmyclass/api/components/classmember/dto/in/ClassMemberInDTO.java (hunk 9 above)
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/classmember/dto/in/ClassMemberInDTO.java]: Should be called **GroupMemberInDTO**

File: src/main/java/com/ohmyclass/api/components/classmember/repository/IClassMemberRepository.java [hunk 10]
```diff
@@ -0,0 +1,13 @@
+package com.ohmyclass.api.components.classmember.repository;
+
+import java.util.Optional;
+
+import org.springframework.data.repository.CrudRepository;
+
+import com.ohmyclass.api.components.classmember.controller.impl.ClassMember;
+
+public interface IClassMemberRepository extends CrudRepository<ClassMember, Long> {
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/api/components/classmember/repository/IClassMemberRepository.java]: Rename to IGroupMemberRepository

File: src/main/java/com/ohmyclass/api/components/classmember/dto/out/ClassMemberOutDTO.java (hunk 8 above)
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/classmember/dto/out/ClassMemberOutDTO.java]: Should be called **GroupMemberOutDTO**

File: src/main/java/com/ohmyclass/api/components/classmember/service/crud/IClassMemberService.java [hunk 11]
```diff
@@ -0,0 +1,3 @@
+package com.ohmyclass.api.components.classmember.service.crud;
+
+public interface IClassMemberService {}
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/api/components/classmember/service/crud/IClassMemberService.java]: Rename to IGroupMemberService

File: src/main/java/com/ohmyclass/api/components/classmember/entity/ClassMember.java (hunk 7 above)
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/classmember/entity/ClassMember.java:17]: Should be called **GroupMember**

File: src/main/java/com/ohmyclass/api/components/classmember/service/crud/impl/ClassMemberService.java [hunk 12]
```diff
@@ -0,0 +1,3 @@
+package com.ohmyclass.api.components.classmember.service.crud.impl;
+
+public class ClassMemberService {}
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/api/components/classmember/service/crud/impl/ClassMemberService.java]: Rename to GroupMemberService

File: src/main/java/com/ohmyclass/api/components/classmember/repository/IClassMemberRepository.java (hunk 10 above)
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/classmember/repository/IClassMemberRepository.java]: Should be called **IGroupMemberRepository**

File: src/main/java/com/ohmyclass/api/components/classmember/service/mapper/AClassMemberMapper.java [hunk 13]
```diff
@@ -0,0 +1,5 @@
+package com.ohmyclass.api.components.classmember.service.mapper;
+
+public abstract class AClassMemberMapper {
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/api/components/classmember/service/mapper/AClassMemberMapper.java]: Rename to AGroupMemberMapper

File: src/main/java/com/ohmyclass/api/components/classmember/service/crud/impl/ClassMemberService.java (hunk 12 above)
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/classmember/service/crud/impl/ClassMemberService.java]: Should be called **GroupMemberService**

File: src/main/java/com/ohmyclass/api/components/classmember/service/crud/IClassMemberService.java (hunk 11 above)
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/classmember/service/crud/IClassMemberService.java]: Should be called **IGroupMemberService**

File: src/main/java/com/ohmyclass/api/components/classmember/service/mapper/AClassMemberMapper.java (hunk 13 above)
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/classmember/service/mapper/AClassMemberMapper.java]: Should be called **AGroupMemberMapper**

File: src/main/java/com/ohmyclass/api/components/comment/entity/Comment.java [hunk 14]
```diff
@@ -1,3 +1,53 @@
...
+
+	@OneToMany(mappedBy = "fkParentComment")
+	private Set<Comment> Children;
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/comment/entity/Comment.java]: lowercase?

File: src/main/java/com/ohmyclass/api/components/comment/entity/Comment.java [hunk 15]
```diff
@@ -1,3 +1,53 @@
...
+
+	@ManyToOne(fetch = FetchType.LAZY)
+	private User fkUser;
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/comment/entity/Comment.java]: same here with fk

File: src/main/java/com/ohmyclass/api/components/comment/entity/Comment.java [hunk 16]
```diff
@@ -1,3 +1,53 @@
...
+
+	@ManyToOne
+	@JoinColumn(name = "fkTask")
+	@JsonManagedReference
+	private Task fkTasks;
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/comment/entity/Comment.java]: and here

File: src/main/java/com/ohmyclass/api/components/group/entity/Group.java [hunk 17]
```diff
@@ -1,3 +1,50 @@
...
+    @Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	@Column(name = "id", nullable = false)
+	private Long id;
+
+	@Column(name = "name")
+	private String name;
+
+	@OneToMany(fetch = FetchType.EAGER)
+	private Set<ClassMember> classmembers;
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/group/entity/Group.java]: Should be called **groupMembers**

File: src/main/java/com/ohmyclass/api/components/role/entity/Role.java [hunk 18]
```diff
@@ -12,10 +12,9 @@
 @Getter
 @Setter
 @Entity
-@Table(name = "role")
-public class Role {
+@Table(name = "role")public class Role {
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/role/entity/Role.java]: Formatting

File: src/main/java/com/ohmyclass/api/components/role/entity/Role.java [hunk 19]
```diff
@@ -12,10 +12,9 @@
...
 
-	@Id
+@Id
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/role/entity/Role.java]: Formatting

File: src/main/java/com/ohmyclass/api/components/subject/entity/Subject.java [hunk 20]
```diff
@@ -1,3 +1,42 @@
...
+	private String name;
+
+	@ManyToMany(fetch = FetchType.EAGER)
+	private List<User> teachers;
+
+	@Column(name = "color", length = 6)
+	private String color;
+
+	@Column(name = "fkGroup")
+	private Group fkGroup;
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/subject/entity/Subject.java]: Remove fk from attribute name but not from column name

File: src/main/java/com/ohmyclass/api/components/subject/entity/Subject.java [hunk 21]
```diff
@@ -1,3 +1,42 @@
...
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	@Column(name = "id", nullable = false)
+	private Long id;
+
+	@Column(name = "name")
+	private String name;
+
+	@ManyToMany(fetch = FetchType.EAGER)
+	private List<User> teachers;
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/subject/entity/Subject.java]: Is it necessary to have multiple teachers per subject?

File: src/main/java/com/ohmyclass/api/components/task/entity/Task.java [hunk 22]
```diff
@@ -1,3 +1,69 @@
...
+	
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	@Column(name = "id", nullable = false)
+	private Long id;
+
+	@Column(name = "title")
+	private String title;
+
+	@Column(name = "Description")
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/task/entity/Task.java]: lowercase

File: src/main/java/com/ohmyclass/api/components/task/entity/Task.java [hunk 23]
```diff
@@ -1,3 +1,69 @@
...
+	@Column(name = "createDate")
+	private Date createDate;
+
+	@Column(name = "editDate")
+	private Date editDate;
+
+	@OneToOne
+	@JoinColumn(name = "fkUser")
+	@JsonManagedReference
+	private User fkUser;
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/task/entity/Task.java]: Remove fk from attribute name

File: src/main/java/com/ohmyclass/api/components/task/entity/Task.java [hunk 24]
```diff
@@ -1,3 +1,69 @@
...
+
+	@OneToMany(fetch = FetchType.LAZY)
+	private List<Comment> comments;
+
+	@ManyToOne
+	@JoinColumn(name = "fkSubject")
+	@JsonManagedReference
+	private Subject fkSubject;
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/task/entity/Task.java]: Same here

File: src/main/java/com/ohmyclass/api/components/task/entity/Task.java [hunk 25]
```diff
@@ -1,3 +1,69 @@
...
+
+	@ManyToOne
+	@JoinColumn(name = "fkGroup")
+	@JsonManagedReference
+	private Group fkGroup;
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/task/entity/Task.java]: And here too

File: src/main/java/com/ohmyclass/api/components/task/repository/ITaskRepository.java [hunk 26]
```diff
@@ -1,5 +1,22 @@
...
+import com.ohmyclass.api.components.task.entity.Task;
+
+@Repository
+public interface ITaskRepository extends CrudRepository<Task, Long> {
+
+	Optional<Task> findTaskById(Long id);
+	
+	Optional<Task> getTaskByGroup(Group group);
+
+    Optional<Task> findTaskByDate(Date Date);
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/task/repository/ITaskRepository.java]: Formatting

File: src/main/java/com/ohmyclass/api/components/user/entity/User.java [hunk 27]
```diff
@@ -33,9 +40,40 @@ public class User {
...
+	@JoinColumn(name = "fkComment", referencedColumnName = "id")
+	private List<Comment> comments;
+
 	@OneToOne(mappedBy = "fkUser")
 	private Preferences preferences;
 
+	@ManyToMany
+	@JoinColumn(name = "fkClassMember")
+	@JsonManagedReference
+	private Set<ClassMember> fkClassMembers;
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/user/entity/User.java]: Remove fk from attribute name

File: src/main/java/com/ohmyclass/api/components/user/entity/User.java [hunk 28]
```diff
@@ -33,9 +40,40 @@ public class User {
...
+
+	@OneToMany
+	@JoinColumn(name ="fkComment")
+	@JsonManagedReference
+	private Set<Comment> fkComments;
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/user/entity/User.java]: here too

File: src/main/java/com/ohmyclass/api/components/user/entity/User.java [hunk 29]
```diff
@@ -33,9 +40,40 @@ public class User {
...
+
+	@ManyToMany
+	@JoinColumn(name = "fkSubject")
+	@JsonManagedReference
+	private Set<Subject> fkSubjects;
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/user/entity/User.java]: here too

File: src/main/java/com/ohmyclass/api/components/user/entity/User.java [hunk 30]
```diff
@@ -33,9 +40,40 @@ public class User {
...
+
+	@OneToOne
+	@JoinColumn(name = "fkTask")
+	@JsonManagedReference
+	private Task fkTask;
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/user/entity/User.java:76]: and here too

File: src/main/java/com/ohmyclass/api/components/user/entity/User.java [hunk 31]
```diff
@@ -33,9 +40,40 @@ public class User {
...
+
+	@OneToOne(cascade = {CascadeType.ALL},
+			orphanRemoval = true,
+			mappedBy = "fkUser")
+	@JsonBackReference
+	private Task task ;
```
Comment (Lezurex) [inline src/main/java/com/ohmyclass/api/components/user/entity/User.java]: One space before semicolon

**Review by Lezurex — CHANGES_REQUESTED** [review]
Some comments from an absolutely dumb person.

In general: Consider removing the **fk** from the attribute names, but not from the column name.

File: src/main/java/com/ohmyclass/api/components/comment/repository/ICommentRepository.java [hunk 32]
```diff
@@ -1,5 +1,17 @@
...
+import org.springframework.stereotype.Repository;
+
+import com.ohmyclass.api.components.comment.entity.Comment;
+
+@Repository
+public interface ICommentRepository extends CrudRepository<Comment, Long> {
+
+    Optional<Comment> findCommentById(Long id);
+
+    Optional<Comment> findCommentOrderByDate(Date date);
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/api/components/comment/repository/ICommentRepository.java]: Order what? A single comment? What are you "querying" for?
And why do you need a date? It's order by date; Not "findByDate".

Change to something like "List<Comment> findAllCommentsOrderByDate();"
But this isn't a good approach either. User "findAllCommentsByClassIdOrderByDate(Long classId)" as an inspiration.

File: src/main/java/com/ohmyclass/api/components/group/entity/Group.java [hunk 33]
```diff
@@ -1,3 +1,50 @@
...
+
+	@ManyToMany(fetch = FetchType.EAGER)
+	private Set<Subject> subjects;
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/api/components/group/entity/Group.java:39]: Change to mc:1. Does a subject have multiple groups? No.

File: src/main/java/com/ohmyclass/api/components/group/repository/IGroupRepository.java [hunk 34]
```diff
@@ -1,5 +1,16 @@
...
+import org.springframework.stereotype.Repository;
+
+import com.ohmyclass.api.components.group.entity.Group;
+
+@Repository
+public interface IGroupRepository extends CrudRepository<Group, Long> {
+
+	Optional<Group> findGroupById(Long id);
+
+	Optional<Group> getAllGroupsByUserId(Long userId);
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/api/components/group/repository/IGroupRepository.java]: findAllGroupsByUserId(Long userId)

File: src/main/java/com/ohmyclass/api/components/subject/entity/Subject.java (hunk 21 above)
Comment (Z-100) [reply src/main/java/com/ohmyclass/api/components/subject/entity/Subject.java]: It's not. 
And why is it a reference to "User"? Groups, nor subjects have direct references to User.
Change to "ClassMember".

File: src/main/java/com/ohmyclass/api/components/subject/repository/ISubjectRepository.java [hunk 35]
```diff
@@ -1,5 +1,16 @@
...
+import org.springframework.data.repository.CrudRepository;
+
+import com.ohmyclass.api.components.group.entity.Group;
+import com.ohmyclass.api.components.subject.entity.Subject;
+
+public interface ISubjectRepository extends CrudRepository<Subject, Long> {
+
+    Optional<Subject> findSubjectById(Long id);
+
+    Optional<Subject> getSubjectByGroup(Group group);
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/api/components/subject/repository/ISubjectRepository.java]: The "standard" you've set for queries is to find entities by ...-ids.
Also: I'm not sure by the "get".
Change to findSubjectByGroupId(Long groupId)

File: src/main/java/com/ohmyclass/api/components/task/entity/Task.java (hunk 23 above)
Comment (Z-100) [reply src/main/java/com/ohmyclass/api/components/task/entity/Task.java]: Should a task be linked to a user directly?
Woulnd't it make more sense to link it to a groupmember?
Tasks are group specific.

File: src/main/java/com/ohmyclass/api/components/task/repository/ITaskRepository.java [hunk 36]
```diff
@@ -1,5 +1,22 @@
...
+
+import com.ohmyclass.api.components.group.entity.Group;
+import com.ohmyclass.api.components.task.entity.Task;
+
+@Repository
+public interface ITaskRepository extends CrudRepository<Task, Long> {
+
+	Optional<Task> findTaskById(Long id);
+	
+	Optional<Task> getTaskByGroup(Group group);
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/api/components/task/repository/ITaskRepository.java]: find. not get.
Also: Change to groupId, to match our repo-standard.

File: src/main/java/com/ohmyclass/api/components/tick/entity/Tick.java [hunk 37]
```diff
@@ -1,3 +1,32 @@
...
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	@Column(name = "id", nullable = false)
+	private Long id;
+
+	@OneToOne(fetch = FetchType.EAGER)
+	private Task task;
+
+	@OneToOne(fetch = FetchType.EAGER)
+	private User user;
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/api/components/tick/entity/Tick.java:33]: User does not have a reference to tick yet.

File: src/main/java/com/ohmyclass/api/components/tick/entity/Tick.java [hunk 38]
```diff
@@ -1,3 +1,32 @@
...
+@Entity
+@Table(name = "tick")
+public class Tick {
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	@Column(name = "id", nullable = false)
+	private Long id;
+
+	@OneToOne(fetch = FetchType.EAGER)
+	private Task task;
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/api/components/tick/entity/Tick.java:30]: Task does not have a reference to tick yet.
And also: Search for "onetoone jpa" on google. I think this annotation is missing some important values.

File: src/main/java/com/ohmyclass/api/components/tick/repository/ITickRepository.java [hunk 39]
```diff
@@ -1,5 +1,13 @@
...
-public interface ITickRepository {
+import java.util.Optional;
+
+import org.springframework.data.repository.CrudRepository;
+
+import com.ohmyclass.api.components.tick.entity.Tick;
+
+public interface ITickRepository extends CrudRepository<Tick, Long>  {
+    
+    Optional<Tick> findTickById(Long id);
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/api/components/tick/repository/ITickRepository.java]: remove unecessary line break

File: src/main/java/com/ohmyclass/util/inteceptor/AccessInterceptor.java [hunk 40]
```diff
@@ -0,0 +1,60 @@
+package com.ohmyclass.util.inteceptor;
```
Comment (Z-100) [inline src/main/java/com/ohmyclass/util/inteceptor/AccessInterceptor.java]: Why does this interceptor exist?
Delete it. (Only util.interceptor.AccessInterceptor; Keep the @Deprecated security.util.interceptor.AccessInterceptor).

**Review by Z-100 — CHANGES_REQUESTED** [review]
Feedback in comments accross the files.

File: src/main/java/com/ohmyclass/api/components/subject/entity/Subject.java (hunk 21 above)
Comment (Lezurex) [reply src/main/java/com/ohmyclass/api/components/subject/entity/Subject.java]: String

File: src/main/java/com/ohmyclass/api/components/task/entity/Task.java (hunk 23 above)
Comment (Lezurex) [reply src/main/java/com/ohmyclass/api/components/task/entity/Task.java]: Rename to something more suitable.

File: src/main/java/com/ohmyclass/util/inteceptor/AccessInterceptor.java (hunk 40 above)
Comment (Z-100) [reply src/main/java/com/ohmyclass/util/inteceptor/AccessInterceptor.java]: Change every single "FetchType.EAGER" to "FetchType.LAZY"
!

**Review by Z-100 — APPROVED** [review]
lgtm

**Pull request merged**